openstack/dns/v2/nameservers/requests.go

```diff
@@ -0,0 +1,45 @@
+package nameservers
+
+import (
+	golangsdk "github.com/opentelekomcloud/gophertelekomcloud"
+)
+
+type ListOpts struct {
+	Hostname *string `q:"hostname,omitempty"`
+	Priority *int    `q:"priority,omitempty"`
+	ZoneID   *string `q:"zone_id,omitempty"`
+}
+
+type ListNameserversRequest struct {
+	Name string `json:"name,omitempty"`
+}
+
+type ListOptsBuilder interface {
+	ToNameserversListQuery() (string, error)
+}
+
+// ToNameserversListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToNameserversListQuery() (string, error) {
+	q, err := golangsdk.BuildQueryString(opts)
+	if err != nil {
+		return "", err
+	}
+	return q.String(), err
+}
+
+func List(client *golangsdk.ServiceClient, zoneID string, opts ListOptsBuilder) (r GetResult) {
+	url := client.ServiceURL("zones", zoneID, "nameservers")
```

Comment:
maybe it should be in `urls.go`

Reply:
is it worth a separate file?
it was in urls.go, but I decided to leave it here

Reply:
as you wish)